lib/surface/compiler/node_translator.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Surface.Compiler.NodeTranslator do
+  @type parse_metadata :: %{line: non_neg_integer(), column: non_neg_integer(), file: binary()}
+
+  @type tag_info :: {:tag_open, binary, list(), parse_metadata(), context()}
```

Comment:
This can also be `{:block_open, ...}`. 

We need a name that can represent both, `tag` and `block` and rename the state's `tags` field accordingly. Technically, they are still tokens that came from the Tokenizer. Maybe `parent_tokens`, `tokens_stack` or `last_tokens`. Not sure. All terrible names :(

We could also call them just `nodes` but then it might be confusing since we already use `node` for translated nodes in the buffers.

Man, naming is hard :)

Reply:
I went for `token_stack`, though `parent_stack` would also be reasonable to me - that indicates it's a hierarchy a bit better... agreed that naming is hard 😂